orchestra/accounts/forms.py

```diff
@@ -28,7 +30,26 @@ class WorkerForm(forms.ModelForm):
     class Meta:
         model = Worker
         fields = ('slack_username', 'phone')
-        exclude = ('user', 'start_datetime')
+        exclude = ('user', 'start_datetime', 'slack_id')
+
+    def clean_slack_username(self):
+        slack_username = self.cleaned_data.get('slack_username')
+        if slack_username:
+            slack = Slacker(settings.SLACK_EXPERTS_API_KEY)
+            slack_id = slack.users.get_user_id(slack_username)
+            if slack_id is None:
+                raise forms.ValidationError(
+                    'Incorrect Slack Username Provided')
+            self.cleaned_data['slack_id'] = slack_id
+        return self.cleaned_data['slack_username']
+
+    def save(self):
```

Comment:
I think you should model the save method off of https://github.com/unlimitedlabs/crowdsurfing/blob/master/product/accounts/forms.py#L57 with the `commit` flag